src/main/java/frc/robot/subsystems/elevator/Elevator.java

```diff
@@ -54,9 +53,17 @@ public MutableMeasure<Distance> getHeightSetpoint() {
         return inputs.heightSetpoint;
     }
 
+    public boolean atSetpoint() {
+        return inputs.heightSetpoint.isNear(
+                inputs.gripperHeight, ElevatorConstants.THRESHOLD.in(Units.Value));
+    }
+
     public Command setHeight(MutableMeasure<Distance> height) {
-        return runOnce(() -> inputs.heightSetpoint = height)
-                .andThen(run(() -> io.setHeight(height)));
+        return Commands.sequence(
+                runOnce(() -> inputs.heightSetpoint = height),
+                run(io::openStopper).withTimeout(0.3),
+                run(() -> io.setHeight(height)).until(this::atSetpoint),
+                run(io::closeStopper).withTimeout(0.3).withName("set height"));
```

Comment:
The with name should be on the whole command.

Reply:
f0a58fe344f2102956cd28025f99cf8de07e7051